### PR TITLE
improve phase-based plan guidance for Zest Dev specs

### PR DESCRIPTION
**Pull request by nettee**
## Summary
- expand the spec template guidance for `## Plan` with checkbox-based, phase-oriented examples
- align summarize-chat and Zest Dev skill docs around capability-based phases, immediate verification, and test-complete phase exit criteria
- clarify that the final phase can focus on broader verification, regression coverage, and coverage improvements

### CODE REVIEW

File: lib/template/spec.md [hunk 1]
```diff
@@ -21,8 +21,24 @@ created: "{date}"
...
+     - [ ] Phase 2: Bar
+       - [ ] Implement: Bar
+       - [ ] Verify: Bar
+     - [ ] Phase 3: Baz
+       - [ ] Implement: Baz
+       - [ ] Verify: Baz
+     Use a capability-based phase breakdown with reviewable, meaningful increments.
+     Good boundaries align with one user-visible workflow, one subsystem/integration boundary, one migration/rollout step, or one stabilization milestone.
+     Each implementation phase must include implementation + immediate testing/verification.
+     The final phase may focus on overall testing/verification, edge cases, regression coverage, and coverage improvements.
```
[the remark this answers is not in the document]
Comment (nettee) [reply lib/template/spec.md:38]: Updated the integration test expectation to match the new phase-plan guidance and pushed commit d7b98cd. Verified with `pnpm test:local` and `pnpm test:package`.
